app/src/test/java/com/rkm/tasky/util/storage/implementation/SessionStorageImplTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.rkm.tasky.util.storage.implementation
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.preferencesDataStoreFile
+import com.rkm.tasky.testUtils.fake.FakeAndroidKeyStore
+import com.rkm.tasky.util.security.implementation.DataStoreEncryptor
+import com.rkm.tasky.util.storage.model.AuthInfo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.*
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+private const val TEST_DATASTORE_NAME: String = "test_datastore"
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+class SessionStorageImplTest {
+
+    init {
+        FakeAndroidKeyStore.setup
+    }
+
+    private val context: Context = RuntimeEnvironment.getApplication()
+    private val dispatcher = StandardTestDispatcher()
+    private val dataStore: DataStore<Preferences> = PreferenceDataStoreFactory.create(
+        produceFile = {context.preferencesDataStoreFile(TEST_DATASTORE_NAME)}
+    )
```

Comment:
Could it make sense making that a JUnit rule? Right now, you always creating a file but never clearing it up again. With a rule, you could encode that before/after behavior and easily reuse it across tests.